### DISCUSSION TITLE
Adding nr_docs option for non-LLM retrieval of representative documents

### DISCUSSION THREAD

**Discussion by Joe-W-Davies** [MaartenGr/BERTopic#1847] [Ideas]
Hi Maarten,

I would like to retrieve more than three representative documents for my topic clusters. I understand this is possible within the LLM framework using the `nr_docs` parameter, but since I am using a custom LLM API not supported by BERTopic, I would like to be able to generate more than 3 within the `get_representative_topics()` function instead, so I can pass them manually in a prompt. 

Please let me know if this is possible! 

Thanks,
Joe

**Comment by MaartenGr** [comment]
The `.get_representative_documents` does not calculate the representative documents but merely extracts those that were already calculated. You would have to use some internal functions to do this. I believe there are a couple of open issues about this, so it should be quick to find.

**Comment by Joe-W-Davies** [comment]
Thanks for your quick reply!

Having a quick look around, is it possible to find them manually for each topic from the label and predicted probability for that label (e.g. using the output of `topic_model.get_document_info()`? Assuming the top representative documents have the highest probability to be in the label.

**Reply by MaartenGr** [reply]
You could indeed do something like that. Personally, I like to focus on the representative docs through MMR with c-TF-IDF in order to find documents that tend to be closer to the topic representation and to a lesser extend the composition of the cluster. I believe you can find more about that here: https://github.com/MaartenGr/BERTopic/issues/848

**Reply by Joe-W-Davies** [reply]
Great, that looks like exactly what I need.

Thanks for for your help and for producing this great repo :)

**Reply by MaartenGr** [reply]
No problem! Glad I could be of help 😄
